src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -128,6 +131,52 @@ public void OnDataReceived(string jsonRpcPayload)
             }
         }
 
+        private void HandleCommandExecutedRequest(string id, string paramsJson)
+        {
+            if (id == null)
+            {
+                _sendToClient(JsonRpcHelper.CreateErrorResult(id, JsonRpcHelper.ErrorCode.InvalidRequest));
+                return;
+            }
+
+            Contracts.AssertValue(id);
+            Contracts.AssertValue(paramsJson);
+
+            if (!TryParseParams(paramsJson, out CommandExecutedParams commandExecutedParams))
+            {
+                _sendToClient(JsonRpcHelper.CreateErrorResult(id, JsonRpcHelper.ErrorCode.ParseError));
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(commandExecutedParams.Argument))
+            {
+                //TODO: Log argument is null.
```

Comment:
>//TODO: Log argument is null. [](http://example.com/codeflow?start=16&length=29)

What should we do here? 
This should never happen right - should we just return an error - similar to line 147? 

Or we can throw (it will get caught in the caller at line 125 and also return an error?)

Reply:
introducing new error code and handling it.